src/Mod/Path/App/TooltablePyImp.cpp

```diff
@@ -328,6 +328,22 @@ PyObject* ToolPy::copy(PyObject * args)
 }
 
 
+PyObject* ToolPy::fromTemplate(PyObject * args)
+{
+    PyObject *pcObj;
+    if (PyArg_ParseTuple(args, "S", &pcObj)) {

```

Comment:
Is there a reason not to use 's' (lower-case) as we do it everywhere else? Now that FreeCAD is mostly ported to Py3 using lower-case "s" won't cause extra porting effort while upper-case "S" will do:
https://docs.python.org/3/c-api/arg.html (search for bytes object)

Btw, the extra check with PyString_Check is superfluous as the docs say that if it's not a str a TypeError will be raised:
https://docs.python.org/2/c-api/arg.html

For performance reason you should not raise an exception here but instead use PyErr_SetString(PyExc_TypeError, ...) and return 0